tests/bluetooth/addr/src/test_bt_addr_from_str.c

```diff
@@ -0,0 +1,154 @@
+/* Copyright (c) 2022 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/ztest.h>
+
+ZTEST_SUITE(bt_addr_from_str, NULL, NULL, NULL, NULL, NULL);
+
+ZTEST(bt_addr_from_str, test_reject_empty_string)
+{
+	char *addr_str = "";
+	bt_addr_t a;
+
+	zassert_equal(bt_addr_from_str(addr_str, &a), -EINVAL);
```

Comment:
Nice idea with the tests. Perhaps, since these test explicitly for `-EINVAL`, we should add that as the only return value in the documentation of `bt_addr_from_str`? i.e. `@retval -EINVAL ...`